Look at all relations for possible updates

Nyholm opened the pull request:
This is a high priority bugfix. The bug was introduced in [1.0.0-beta20](https://github.com/graphaware/neo4j-php-ogm/commit/491fcbb4cef690732e2ba39ef08cf9e3d2cd9e3f#diff-ffe84599a511ad2c003ae8967ed1b01fR189)

If we have more than one relationship and the first one we check is empty, then no other relations are checked.

ikwattro commented:
Thanks. Can you create a Test proving the issue and the fix, then it can get merged.

Nyholm commented:
Ill try.. I have a hard time running the tests on my local machine

ikwattro commented:
Or just paste here some code that you use, I'll make a test then, no worries

ikwattro commented:
For running the tests, what is the issue actually ?

```bash
ikwattro@Christophes-MacBook-Air ~/d/g/p/ogm> ./vendor/bin/phpunit
PHPUnit 5.6.2 by Sebastian Bergmann and contributors.

Runtime:       PHP 7.0.12 with Xdebug 2.4.1
Configuration: /Users/ikwattro/dev/graphaware/php/ogm/phpunit.xml

................................................................. 65 / 78 ( 83%)
.............                                                     78 / 78 (100%)

Time: 8.31 seconds, Memory: 22.00MB

OK (78 tests, 402 assertions)
ikwattro@Christophes-MacBook-Air ~/d/g/p/ogm>
```

Nyholm commented:
How come they fail on Travis?

I'm on my way from the office now, I'll try tomorrow. I remember I had some issues (external requirements) but it might have been on the php client and not the ogm

ikwattro commented:
The fail on Travis was due to another PR merged, it is fixed now

ikwattro commented:
For the requirements, yeah 5.6 and the requirements of the client : mbstring and bcmath extensions. Note that chances are high that the first official release of the ogm will be 7+ only.

ikwattro commented:
@Nyholm your last commit "ref the case" is fixed already in master, just rebase.

Nyholm commented:
I found the issue why I was having problems running the test locally. I fixed those in #35 

I've added tests for this PR. I also found a bug where the namespaces had wrong case

Nyholm commented:
Thanks. I've rebased

ikwattro commented:
Thx

Nyholm commented:
Thank you for merging
